test/src/test/java/org/corfudb/runtime/collections/CorfuQueueTest.java

```diff
@@ -270,4 +285,37 @@ public void queueCheckpointTest() {
                     .isEqualTo(String.valueOf(itemIdx));
         });
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void logUpdateThatLooksLikeEnqueue() {
+        final CorfuStore corfuStore = new CorfuStore(createDefaultRuntime());
+        final String namespace = "some_ns";
+        final String tableName = "some_table";
+        Table<Queue.CorfuGuidMsg, RoutingTableEntryMsg, Queue.CorfuQueueMetadataMsg> q = null;
+        try {
+            q = corfuStore.openQueue(namespace, tableName, RoutingTableEntryMsg.class,
+                    TableOptions.fromProtoSchema(RoutingTableEntryMsg.class));
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+        final Table<Queue.CorfuGuidMsg, RoutingTableEntryMsg, Queue.CorfuQueueMetadataMsg> finalQ = q;
+        final byte[] foo = {'a', 'b', 'c'};
+        executeTxn(corfuStore, namespace, (TxnContext tx) -> tx.enqueue(finalQ,
+                RoutingTableEntryMsg.newBuilder().setOpaquePayload(ByteString.copyFrom(foo)).build()));
+
+        Object[] smrArgs = new Object[2];
+        smrArgs[0] = Queue.CorfuGuidMsg.newBuilder().setInstanceId(1).build();
+        smrArgs[1] = RoutingTableEntryMsg.newBuilder().setOpaquePayload(ByteString.copyFrom(foo)).build();
+        executeTxn(corfuStore, namespace, (TxnContext tx) -> tx.logUpdate(finalQ.getStreamUUID(),
+                new SMREntry("put", smrArgs, corfuStore.getRuntime().getSerializers().getSerializer(ProtobufSerializer.PROTOBUF_SERIALIZER_CODE))
+        ));
+
+        // validate if logUpdate and enqueu produces same result
+        executeTxn(corfuStore, namespace, (TxnContext tx) -> {
+            //RoutingTableEntryMsg entry = tx.entryList(finalQ).get(0).getEntry();
+            assertThat(tx.entryList(finalQ).get(0).getEntry()).isEqualTo(foo);
```

Comment:
Do we need to change this line as `assertThat(tx.entryList(finalQ).get(0).getEntry()).isEqualTo(tx.entryList(finalQ).get(1).getEntry());`
? Since in my PR, after doing enqueue and logUpdate, entryList of finalQ should have 2 elements. The first element is the result of enqueue, and the second element is the result of logUpdate. Changing it can compare the result of enqueue and logUpdate.

Reply:
Exactly! yes we need that line to compare that the two operations produce the same result

Reply:
*line 317

Reply:
Sundar, are we going to change this assertion by comparing the 2 arguments?